CMakeLists.txt

```diff
@@ -42,6 +42,21 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Build Halide with ccache if the package is present
+set(Halide_CCACHE_BUILD OFF CACHE BOOL "Set to ON for a ccache enabled build")
+if (Halide_CCACHE_BUILD)
+  find_program(CCACHE_PROGRAM ccache)
+  if (CCACHE_PROGRAM)
+      set(Halide_CCACHE_PARAMS "CCACHE_CPP2=yes CCACHE_HASHDIR=yes"
+          CACHE STRING "Parameters to pass through to ccache")
+      set(CCACHE_PROGRAM "${Halide_CCACHE_PARAMS} ${CCACHE_PROGRAM}")
+      set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_PROGRAM})
```

Comment:
```
set(CMAKE_CXX_COMPILER_LAUNCHER "${Halide_CCACHE_PARAMS} ${CCACHE_PROGRAM}")
```
I prefer not to set _global_ properties that would affect an `add_subdirectory` user when a project-local variable would suffice.

Reply:
Also, this is broken on Windows since it's using bash/ksh syntax. This is portable:

```
set(CMAKE_CXX_COMPILER_LAUNCHER "${CMAKE_COMMAND} -E env ${Halide_CCACHE_PARAMS} ${CCACHE_PROGRAM}")
```